include/deal.II/base/thread_management.h

```diff
@@ -1005,8 +1007,98 @@ namespace Threads
     Task(const std::function<RT()> &function_object)
     {
       if (MultithreadInfo::n_threads() > 1)
-        task_data = std::make_shared<TaskData>(
-          std::async(std::launch::async, function_object));
+        {
+#  ifdef DEAL_II_WITH_TBB
+          // Create a promise object and from it extract a future that
+          // we can use to refer to the outcome of the task. For reasons
+          // explained below, we can't just create a std::promise object,
+          // but have to make do with a pointer to such an object.
```

Comment:
```suggestion
          // but have to maintain a (unique) pointer to such an object.
```
(I think "to maintain" is better understood than "to make do" in this context, even though I fully appreciate the sentiment expressed by "to make do" :smile:)